src/components/Select/Select.tsx

```diff
@@ -305,7 +308,8 @@ export const Select = React.forwardRef<HTMLButtonElement, SelectProps>(function
         <div
             ref={controlWrapRef}
             className={selectBlock(mods, className)}
-            {...focusWithinProps}
+            onFocus={mobile ? onFocus : focusWithinProps.onFocus}
+            onBlur={mobile ? onBlur : focusWithinProps.onBlur}
```

Comment:
How is this different from disabling the hook? @korvin89 suggested the following
```js
    const {focusWithinProps} = useFocusWithin({
        onFocusWithin: onFocus,
        onBlurWithin: React.useCallback(
            (e: React.FocusEvent) => {
                onBlur?.(e);
                if (!mobile) {
                    // close on blur only on desktop
                    handleClose();
                }
            },
            [handleClose, onBlur],
        ),
    });
```